integration/tests/cook/test_basic.py

```diff
@@ -67,6 +67,53 @@ def test_basic_submit(self):
         else:
             self.logger.info(f'Exit code not checked because cook executor was not used for {instance}')
 
+    def test_submit_requesting_gpus(self):
+        settings_dict = util.settings(self.cook_url)
+        gpu_enabled = settings_dict['mesos-gpu-enabled']
+        if not gpu_enabled:
+            self.skipTest("Cluster is not gpu-enabled")
+            self.logger.info("Cluster is not gpu-enabled")
+        else:
+            valid_gpu_models_config_map = settings_dict.get("pools", {}).get("valid-gpu-models", [])
+            if not valid_gpu_models_config_map:
+                self.skipTest("No pools have valid-gpu-models")
+                self.logger.info("No pools have valid-gpu-models")
+            else:
+                active_pools, _ = util.active_pools(self.cook_url)
+                if len(active_pools) == 0:
```

Comment:
I would probably turn this into a list comprehension. I.e., make a list of all pools that have a matching valid model. If the list is empty, skip the whole test. If not, run the test.That way you don't log 'pool does not support GPUs'  for the unused pools, and if the test has no valid pools, the test appears as skipped (instead of passed, without actually testing anything)

Reply:
I like that idea thanks! Will implement

Reply:
implemented